meta-lxatac-bsp: lxatac-factory-data: order service before udev

hnez opened the pull request:
This is an altternative to #107. Since both approaches are valid and have their benefits and drawbacks I made this a competing PR.

Order the factory data collecting service before the udev systemd service to make sure that the `.link` file it creates exists before udev starts looking for rules.

This should fix an issue we have that results in the `tac-bridge` interface having the wrong MAC address on some boots.

In contrast to #107 it does so without triggering udev to reload its rules.

hnez committed:
meta-lxatac-bsp: lxatac-factory-data: run script before udev starts

This makes sure that our tac-bridge.link file is ready before udev
reads it. We have previously run into race conditions where the
file was not present at the correct point in time, resulting in the
bridge interface coming up with the wrong MAC address.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

michaelolbrich commented on meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.service:
```diff
@@ -1,10 +1,13 @@
+DefaultDependencies=no
+After=systemd-remount-fs.service
+Before=systemd-udevd.service
+RequiresMountsFor=/run
```
`/run` is always mounted before any service starts, so this line is unnecessary.

hnez replied:
I did a quick `rg RequiresMountsFor` before adding that to check if other service files in the wild had it and there were some.
But if we do not need it that's even better.

Dropped the `RequiresMountsFor=`.

michaelolbrich commented on meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh:
```diff
@@ -67,4 +65,3 @@ OriginalName=tac-bridge
 MACAddress=${MAC_BRIDGE}
 MACAddressPolicy=none
 EOF
-
```
Unrelated change.

hnez replied:
Split into a separate commit.

michaelolbrich commented on meta-lxatac-bsp/recipes-core/lxatac-factory-data/files/lxatac-factory-data.sh:
```diff
@@ -11,14 +11,12 @@ DST_LINK_FILE_BASE="/run/systemd/network/"
 then
-    hostname="$(hostnamectl --transient hostname)"
-    hostnamectl --static hostname "${hostname}"
+    hostname > /etc/hostname
```
This change deserves to be mentioned in the commit message.

hnez replied:
Split into a separate commit with some info in the commit message.
Also added more information in the comment.

hnez committed:
meta-lxatac-bsp: lxatac-factory-data: remove trailing whitespace

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-bsp: lxatac-factory-data: remove reliance on hostnamed

We want to pull the exectution of the factory-data script earlier in the boot
process, so it runs before udev starts.
Relying on a DBus service to get and set the hostname goes against that goal.
Just asking the kernel and writing to /etc/hostname is more or less equivalent
to what we have done before, so just do that.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

michaelolbrich reviewed (approved):
The correct spelling is "D-Bus" but in the commit message it does not really matter.
The rest looks good
